docs/contributing-to-airbyte/tutorials/building-a-java-destination.md

```diff
@@ -0,0 +1,175 @@
+* Step 1: Create the destination using the template generator
+* Step 2: Build the newly generated destination 
+* Step 3: Implement `spec` to define the configuration required to run the connector
+* Step 4: Implement `check` to provide a way to validate configurations provided to the connector
+* Step 5: Implement `write` to write data to the destination
+* Step 6: Set up Acceptance Tests
+* Step 7: Write unit tests or integration tests
+* Step 8: Update the docs \(in `docs/integrations/destinations/<destination-name>.md`\)
+
+{% hint style="info" %}
+All `./gradlew` commands must be run from the root of the airbyte project.
+{% endhint %}
+
+{% hint style="info" %}
+If you need help with any step of the process, feel free to submit a PR with your progress and any questions you have, or ask us on [slack](https://slack.airbyte.io).
+{% endhint %}
+
+## Explaining Each Step
+
+### Step 1: Create the destination using the template
+
+Airbyte provides a code generator which bootstraps the scaffolding for our connector.
+
+```bash
+$ cd airbyte-integrations/connector-templates/generator # assumes you are starting from the root of the Airbyte project.
+$ ./generate.sh
+```
+
+Select the `Java Destination` template and then input the name of your connector. We'll refer to the destination as `<name>-destination` in this tutorial, but you should replace `<name>` with the actual name you used for your connector e.g: `BigQueryDestination` or `bigquery-destination`. 
+
+### Step 2: Build the newly generated destination
+
+You can build the destination by running:
+
+```bash
+# Must be run from the Airbyte project root
+./gradlew :airbyte-integrations:connectors:destination-<name>:build
+```
+
+this compiles the java code for your destination and builds a Docker image with the connector. At this point, we haven't implemented anything of value yet, but once we do, you'll use this command to compile your code and Docker image. 
+
+{% hint style="info" %}
+    Airbyte uses Gradle to manage Java dependencies. To add dependencies for your connector, manage them in the `build.gradle` file inside your connector's directory.   
+{% endhint %}
+
+#### Iterating on your implementation
+
+We recommend the following ways of iterating on your connector as you're making changes:
+
+1. Test-driven development in Java 
+2. Test-driven development using Airbyte's Acceptance Tests
+3. Directly running the docker image 
+
+**Test-driven development in Java**
+This should feel like a standard flow for a Java developer: you make some code changes then run java tests against them. You can do this directly in your IDE, but you can also run all unit tests via Gradle by running the command to build the connector: 
+
+```
+./gradlew :airbyte-integrations:connectors:destination-<name>:build
+```
+
+This will build the code and run any unit tests. This approach is great when you are testing local behaviors and writing unit tests. 
+
+**TDD using acceptance tests & integration tests**
```

Comment:
Nit. An acronym is better introduced by putting it in a pair of parentheses when the full term, test-driven development, is first mentioned in this doc:

> Test-driven development (TDD)

Reference: https://blog.apastyle.org/apastyle/abbreviations/#Q2

Reply:
done